Feat/docs (#27)
Kentico/xperience-by-kentico-aira-unified: README.md

## Description

> Aira Unified integration provides an alternative administration UI and chatbot that can be easily added to an Xperience by Kentico (XbyK) project. It is designed for content editors and marketers who need essential functionality on mobile devices.

Reviewer: Remove the bracket with Xbyk We publicly use only Xperience or Xperience by Kentico.

Author: Done

## Screenshots

> ![UI Application](/images/screenshots/ui_application.png)

Author: I have changed it to a correct images.

| ----------------- | -------------------- |

> | >= 30.0.0         | >= 0.1.0-prerelase-1 |

Reviewer 2: Is this requirement of >= v30.0.0 accurate?

Author: At the moment this is accurate. Update to version 32.0.0 will be realeased in another pull request.

## Quick Start

1. Add the Aira Unified API subscription key to your `appsettings.json`:

```json
"AiraUnifiedOptions": {
  "AiraUnifiedApiSubscriptionKey": "<your aira unified API key>"
}
```
2. Configure your project for [HTTPS](https://learn.microsoft.com/en-us/aspnet/core/security/enforcing-ssl).

3. Include the `Kentico.Xperience.AiraUnified` package in your project:

   ```powershell
   dotnet add package Kentico.Xperience.AiraUnified
   ```

4. Register the required services in `Program.cs`:

   ```csharp
   var builder = WebApplication.CreateBuilder(args);

   // ...
   
   builder.Services.AddKenticoAiraUnified(builder.Configuration);
   
   // ...

   var app = builder.Build();

   // ...
   
   app.UseAiraUnifiedEndpoints();
   
   app.Run();
   ```

5. Configure the Aira Unified settings in the administration UI:
   - **Relative Path Base**: Defines where Aira Unified is available.
   - **Logo**: Select an asset from the Media Library.
   - **Chat Title**: Title for the chat page.
   - **Smart Upload Title**: Title for the smart upload page.
   
   ![Admin Configuration](/images/AiraUnifiedAdminConfiguration.png)

> 6. Set up a `Content Type` for [Mass Asset Upload](https://docs.kentico.com/developers-and-admins/development/content-types#mass-asset-upload-configuration).

Reviewer 2: The ACA will not work with the Dancing Goat sample app without customization because Dancing Goat (at least in v30.2.0) has 2 workspaces - "Dancing Goat Events" and "Dancing Goat Ltd.". The ACA expects a workspace named "Kentico Default" (code name `KenticoDefault`).

This limitation also applies to any app under development that doesn't have a "Kentico Default" workspace.

We should either document that users must create a workspace with a code name `KenticoDefault` or we allow the user to define a "Default" workspace in the the AIRA Unified settings in the Xperience administration.

Author: For now, I have documented the need for the `KenticoDefault` workspace and we will discuss internally how to address this issue.

> 7. Configure role-based permissions in the `Role Management` application:

Reviewer 2: This needs to mention the administrator user _must_ be assigned to a custom role with the correct permissions.

In the Xperience administration, the administrator user/role has access to everything and that's what users will expect with the ACA.

Author: I have added a comment to the `README.md` file, but the fix is already scheduled and will be released in another PR.

   - **View**: Access the Aira Unified chat.
   - **Create/Update**: Access and upload content via Smart Upload.

   ![Role Configuration](/images/ConfigureAiraUnifiedPermissions.png)

> 8. Users can now access Aira Unified at the specified path.

Reviewer 2: ```suggestion
8. Users can now sign in to the Aira Unified app - `<your-path-base>/signin`
```

^ We need to be explicit about the signin requirement here because the ACA does not yet redirect users to `<your-path-base>/signin` if they are not authenticated. Instead they'll get a 404 of the website channel configured for the domain of the ACA request.

Can we pre-populate the path base value with something like `/aira-unified` to make the setup easier?

Author: I have changed the documentation line to your suggestion.
The pre-popilation of the path base makes sense to me, but we need to discuss it internally first.

## Quick Start

1. Add the Aira Unified API subscription key to your `appsettings.json`:

```json
"AiraUnifiedOptions": {
  "AiraUnifiedApiSubscriptionKey": "<your aira unified API key>"
}
```
2. Configure your project for [HTTPS](https://learn.microsoft.com/en-us/aspnet/core/security/enforcing-ssl).

3. Include the `Kentico.Xperience.AiraUnified` package in your project:

   ```powershell
   dotnet add package Kentico.Xperience.AiraUnified
   ```

4. Register the required services in `Program.cs`:

   ```csharp
   var builder = WebApplication.CreateBuilder(args);

   // ...
   
   builder.Services.AddKenticoAiraUnified(builder.Configuration);
   
   // ...

   var app = builder.Build();

   // ...
   
   app.UseAiraUnifiedEndpoints();
   
   app.Run();
   ```

> 5. Configure the Aira Unified settings in the administration UI:

Author: I have added clearer explanation

   - **Relative Path Base**: Defines where Aira Unified is available.
   - **Logo**: Select an asset from the Media Library.
   - **Chat Title**: Title for the chat page.
   - **Smart Upload Title**: Title for the smart upload page.
   
   ![Admin Configuration](/images/AiraUnifiedAdminConfiguration.png)

6. Set up a `Content Type` for [Mass Asset Upload](https://docs.kentico.com/developers-and-admins/development/content-types#mass-asset-upload-configuration).
7. Configure role-based permissions in the `Role Management` application:
   - **View**: Access the Aira Unified chat.

>    - **Create/Update**: Access and upload content via Smart Upload.

Author: They are different, With view permission you can chat with AIRA. The create/update permission lets you access the smart uploader and upload images.

>    - **Create/Update**: Access and upload content via Smart Upload.

Reviewer: So you need both create and update permissions to access and upload images?

>    - **Create/Update**: Access and upload content via Smart Upload.

Author: No, I am sorry, this was a bug and I have fixed this already, so the only required permission is the Create. I will change this in the docs.

>    - **Create/Update**: Access and upload content via Smart Upload.

Reviewer: Thanks for the clarification. Please also update the screenshot below because it still shows the Update permission.

>    - **Create/Update**: Access and upload content via Smart Upload.

Author: I have updated the image

   ![Role Configuration](/images/ConfigureAiraUnifiedPermissions.png)

At the moment the administrator user/role must be assigned to a custom role with the correct permission.

> 8. In case of using this library in a project with XbyK versions > 30.2.0 the Aira Unified expects a workspace named "Kentico Default" (code name 'KenticoDefault'). In that case, add a workspace with a code name `KenticoDefault` workspace.

Author: Rephrased

9. Users can now sign in to the Aira Unified app - `<your-path-base>/signin`.

> 

Author: Added

> ## Full Instructions

Author: I have moved it

> 

Author: I have deleted this sentence, because the setup is meant for any project, not just the Dancing Goat

7. Configure role-based permissions in the `Role Management` application:
   - **View**: Access the Aira Unified chat.
   - **Create**: Access and upload content via Smart Upload.

   ![Role Configuration](/images/ConfigureAiraUnifiedPermissions.png)

At the moment the administrator user/role must be assigned to a custom role with the correct permission.

8. Aira Unified expects a workspace named "Kentico Default" (code name 'KenticoDefault'). When using this library in a project with Xperience versions > 30.2.0, add a workspace with the code name `KenticoDefault`.

9. Users can now sign in to the Aira Unified app - `<your-path-base>/signin`.

View the [Usage Guide](docs/Usage-Guide.md) for more instructions.

## Contributing

To see the guidelines for Contributing to Kentico open source software, please see [Kentico's `CONTRIBUTING.md`](https://github.com/Kentico/.github/blob/main/CONTRIBUTING.md) for more information and follow the [Kentico's `CODE_OF_CONDUCT`](https://github.com/Kentico/.github/blob/main/CODE_OF_CONDUCT.md).

> Find project-specific contribution details.

Reviewer: Why not leave the link to the contributing setup? I think it would be useful to leave it here.

Author: restored
